src/Groups/GAPGroups.jl

```diff
@@ -1332,6 +1333,128 @@ function relators(G::FPGroup)
    return [group_element(F,L[i]) for i in 1:length(L)]
 end
 
+
+@doc Markdown.doc"""
+    map_word(g::FPGroupElem, genimgs::Vector; genimgs_inv::Vector = Vector(undef, length(genimgs)), init = nothing)
+    map_word(v::Vector{Union{Int, Pair{Int, Int}}}, genimgs::Vector; genimgs_inv::Vector = Vector(undef, length(genimgs)), init = nothing)
+
+Return the product $R_1 R_2 \cdots R_n$
+that is described by `g` or `v`, respectively.
+
+If `g` is an element of a free group $G$, say, then the rank of $G$ must be
+equal to the length of `genimgs`, `g` is a product of the form
+$g_{i_1}^{e_i} g_{i_2}^{e_2} \cdots g_{i_n}^{e_n}$
+where $g_i$ is the $i$-th generator of $G$ and the $e_i$ are nonzero integers,
+and $R_j = $`imgs[`$i_j$`]`$^{e_j}$.
+
+If `g` is an element of a finitely presented group then the result is
+defined as `map_word` applied to a representing element of the underlying
+free group.
+
+If the first argument is a vector `v` of integers $k_i$ or pairs `k_i => e_i`,
+respectively,
+then the absolute values of the $k_i$ must be at most the length of `genimgs`,
+and $R_j = $`imgs[`$|k_i|$`]`$^{\epsilon_i}$
+where $\epsilon_i$ is the `sign` of $k_i$ (times $e_i$).
+
+If a vector `genimgs_inv` is given then its assigned entries are expected
+to be the inverses of the corresponding entries in `genimgs`,
+and the function will use (and set) these entries in order to avoid
+calling `inv` (more than once) for entries of `genimgs`.
+
+If `v` has length zero then `init` is returned if also `genimgs` has length
+zero, otherwise `one(genimgs[1])` is returned.
+In all other cases, `init` is ignored.
+
+# Examples
+```jldoctest
+julia> F = free_group(2);  F1 = gen(F, 1);  F2 = gen(F, 2);
+
+julia> imgs = gens(symmetric_group(4))
+2-element Vector{PermGroupElem}:
+ (1,2,3,4)
+ (1,2)
+
+julia> map_word(F1^2, imgs)
+(1,3)(2,4)
+
+julia> map_word(F2, imgs)
+(1,2)
+
+julia> map_word(one(F), imgs)
+()
+
+julia> invs = Vector(undef, 2);
+
+julia> map_word(F1^-2*F2, imgs, genimgs_inv = invs)
+(1,3,2,4)
+
+julia> invs
+2-element Vector{Any}:
+    (1,4,3,2)
+ #undef
+
+```
+"""
+function map_word(g::FPGroupElem, genimgs::Vector; genimgs_inv::Vector = Vector(undef, length(genimgs)), init = nothing)
+  G = parent(g)
+  Ggens = gens(G)
+  if length(Ggens) == 0
+    return init
+  end
+  gX = g.X
+  if ! GAP.Globals.IsAssocWord(gX)
+    # element of a f.p. group
+    gX = GAP.Globals.UnderlyingElement(gX)
+  end
+  _names = GAP.evalstr("x -> x!.names")
+  @assert length(_names(GAP.Globals.FamilyObj(gX))) == length(genimgs)
+#TODO: improve this, see https://github.com/oscar-system/GAP.jl/issues/771
+  @assert GAP.Globals.IsAssocWord(gX)
+  if GAP.Globals.IsLetterAssocWordRep(gX)
+    ll = Vector{Int}(GAP.Globals.LetterRepAssocWord(gX))
+  elseif GAP.Globals.IsSyllableAssocWordRep(gX)
+    l = ExtRepOfObj(gX)
+    ll = Pairs{Int, Int}[]
```

Comment:
Syntax error:
```suggestion
    ll = Pair{Int, Int}[]
```
This code branch is not covered by a test, which is why this could slip through :-)

Reply:
Yes. I am going to add suitable tests.